### PR TITLE
Modify deadline, modify and add tests

### LINKED ISSUE #80: edit deadline format and modify tests

### PR DESCRIPTION
**Pull request by umergta**
- make HH:mm field optional
- add tests
- modify tests and change constant names
- update documentation

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -80,15 +80,15 @@ Adds a task to ModuleBook3.5.
 Format: `add n/TASK NAME m/MODULE d/DESCRIPTION b/DEADLINE [t/TAG]…​`
 
 <div markdown="span" class="alert alert-primary">:bulb: **Tip:**
-Enter the time you wish to complete your task in the format: YYYY-mm-dd HH:mm
+Enter the time you wish to complete your task in the format: yyyy-MM-dd HH:mm or yyyy-MM-dd (both accepted)
```
Comment (yhtMinceraft1010X) [inline docs/UserGuide.md]: I think the UG should also mention what happens when the time is not keyed in. E.g. The current time on your computer is taken.

**Review by yhtMinceraft1010X — COMMENTED** [review]
Other than the comment about UG, LGTM

**Review by QY-H00 — COMMENTED** [review]
LGTM for the test part!

**Review by AiwassPrime — COMMENTED** [review]
LGTM

File: docs/UserGuide.md (hunk 1 above)
Comment (umergta) [reply docs/UserGuide.md]: ok will do

Comment (umergta) [reply docs/UserGuide.md]: done

**Pull request merged**